api/odl.py

```diff
@@ -890,3 +956,509 @@ def _url_for(self, *args, **kwargs):
         del kwargs["_external"]
         return "http://%s?%s" % ("/".join(args), "&".join(["%s=%s" % (key, val) for key, val in kwargs.items()]))
 
+
+class SharedODLAPI(BaseCirculationAPI):
```

Comment:
A general note for later: a lot of the parsing could be handled by a specialized OPDS parser. This was true before, when we were just parsing metadata out of the OPDS, but now that we're doing sophisticated matches against fulfillment mechanisms to find the right URL it seems a little more important.